q2_types_genomics/reference_db/_format.py

```diff
@@ -5,17 +5,81 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
-
-
+import re
 from qiime2.plugin import model
+from qiime2.core.exceptions import ValidationError
 from q2_types_genomics.plugin_setup import plugin
-from q2_types_genomics.reference_db._type import ReferenceDB, Eggnog, Diamond
+from q2_types_genomics.reference_db._type import (
+    ReferenceDB, Eggnog, Diamond, EggnogSequenceTaxa
+)
+from q2_types.feature_data import MixedCaseProteinFASTAFormat
 
 
 class EggnogRefTextFileFmt(model.TextFileFormat):
+    _expected_columns = [
+            '# Taxid',
+            'Sci.Name',
+            'Rank',
+            'Named Lineage',
+            'Taxid Lineage'
+    ]
+    _line_pattern = re.compile(
+        r'^\d+\t'  # Taxid
+        r'([^\t]*\t)'  # Sci.Name
+        r'(no rank|species|subspecies)\t'  # Rank
```

Comment:
I'm feeling slightly uncomfortable about this one - this assumes that if there is a rank provided in the data, it will always be either species or subspecies - is this really the case? Are there never any ranks higher than those present?

Reply:
those are really the only ones present in the data...

```python
>>> df["Rank"].unique()
array(['no rank', 'species', 'subspecies'], dtype=object)
```

Reply:
OK, thanks for checking! We can always update in the future if anything changes 🙌